### DISCUSSION TITLE
expose the call api of tenacity

### DISCUSSION THREAD

**Discussion by RonnyPfannschmidt** [hynek/stamina#45] [Ideas]
under https://tenacity.readthedocs.io/en/latest/#changing-arguments-at-run-time

it is shown how to do a retry call where the function configures the retrying but will not have to boilerplate the context manager/attempt loop for a custom callable

i'd love to see that api passed trough

unfortunately it looks like stamina wraps only around a argumented AsyncRetrying instead of all supported retrying objects

**Comment by hynek** [comment]
Could you please outline what you want to actually do, without referencing concrete Tenacity features? What problem do you want solved?

**Reply by RonnyPfannschmidt** [reply]
currently stamina only provides `stamina.retry_context`
this requires quite some extra code for invoking a function with the retry configuration

```
#instead of
for attempt in stamina.retry_context(on=httpx.HTTPError):
    with attempt:
       res = myfunc(...)
       
# i want

retryer = stamina.Retryer(on=httpx.HTTPError)
res = retryer(myfunc, ...)

```

**Comment by hynek** [comment]
Everybody interested in this please head over to #56
